src/handler/unarchive.ts

```diff
@@ -0,0 +1,86 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda/trigger/api-gateway-proxy';
+import 'dotenv/config';
+import { cloneDeep } from 'lodash';
+import { TechRecordType } from '@dvsa/cvs-type-definitions/types/v3/tech-record/tech-record-verb';
+import { SearchCriteria } from '../models/search';
+import { setCreatedAuditDetails } from '../services/audit';
+import { getBySystemNumberAndCreatedTimestamp, postTechRecord, searchByCriteria } from '../services/database';
+import { getUserDetails } from '../services/user';
+import { StatusCode } from '../util/enum';
+import { formatTechRecord } from '../util/formatTechRecord';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { validateUnarchiveErrors } from '../validators/unarchive';
+import { UnarchiveRequestBody } from '../models/unarchive';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Unarchive end point called');
+
+  try {
+    const unarchiveErrors = validateUnarchiveErrors(event);
+
+    if (unarchiveErrors) {
+      return addHttpHeaders(unarchiveErrors);
+    }
+
+    const systemNumber = decodeURIComponent(event?.pathParameters?.systemNumber ?? '');
+    const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp ?? '');
+    const userDetails = getUserDetails(event.headers.Authorization ?? '');
+    const { reasonForUnarchiving, status } = JSON.parse(event.body as string) as UnarchiveRequestBody;
+
+    logger.info(`Get from database with systemNumber ${systemNumber} and timestamp ${createdTimestamp}`);
+
+    const archivedRecord: TechRecordType<'get'> = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp);
+    logger.debug(`result is: ${JSON.stringify(archivedRecord)}`);
+
+    if (!archivedRecord || !Object.keys(archivedRecord).length) {
+      return addHttpHeaders({
+        statusCode: 404,
+        body: `No record found matching systemNumber ${systemNumber} and timestamp ${createdTimestamp}`,
+      });
+    }
+
+    if (archivedRecord.techRecord_statusCode !== StatusCode.ARCHIVED) {
+      return addHttpHeaders({
+        statusCode: 400,
+        body: 'Record provided is not an archived record so cannot be unarchived.',
+      });
+    }
+
+    const allVehicleRecords = await searchByCriteria(SearchCriteria.SYSTEM_NUMBER, archivedRecord.systemNumber);
+    const hasUnarchivedRecords: boolean = allVehicleRecords.some((searchResult) => searchResult.techRecord_statusCode !== StatusCode.ARCHIVED
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```

Comment:
Can we see if we can clean up the any here? Should be okay now we have proper types

Also, not sure I get why we need to check the vrm?

Reply:
The requirements state that we have to check for all associated tech records that share the same VRM. The Get type doesn't support the primaryVrm field, though it probably should.

Reply:
The get doesn't have primaryVrm!?? That has to be a mistake